### PR TITLE
Finish schema

### LINKED ISSUE #41: Create IncidentContacts
implement many to many relationship

### LINKED ISSUE #42: Create ContactRoles
Implement many to many relationship

### LINKED ISSUE #43: Create Roles Table
Create roles table

### PR DESCRIPTION
**Pull request by amsmyth1**
### What Changed?
- Add Roles, ContactRoles and IncidentContacts to the schema
- Add relationships between models, all tested
### What's Next?
- Contact CRUD Functionality
### Known Issues?
- none
### Test Coverage?
- 100%
### README & Documentation
- No updates for this pull 

### Issues Closed 
Close #41 
Close #42 
Close #43

### CODE REVIEW

File: db/migrate/20210522111400_create_incident_contacts.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class CreateIncidentContacts < ActiveRecord::Migration[5.2]
+  def change
+    create_table :incident_contacts do |t|
+      t.references :incident, foreign_key: true
+      t.references :contact, foreign_key: true
```
Comment (klatour324) [inline db/migrate/20210522111400_create_incident_contacts.rb:5]: Just double checking, are we to also have title in our `incident_contacts` table?

**Review by klatour324 — COMMENTED** [review]
Looks good Alexa. I just had one comment in regards to our `incident_contacts` table.

File: db/migrate/20210522111400_create_incident_contacts.rb (hunk 1 above)
Comment (amsmyth1) [reply db/migrate/20210522111400_create_incident_contacts.rb:5]: Great catch!

**Review by klatour324 — APPROVED** [review]
Looks good!

**Review by jmichlitsch — APPROVED** [review]

**Pull request merged**